### DISCUSSION TITLE
Multiple Processors Using the Same Data?

### DISCUSSION THREAD

**Discussion by fsmMLK** [wwkimball/yamlpath#165] [Ideas]
I tried using your module to manage yaml files. However there is no place that shows how to do that. I was able to create the processor but after that what? How do I access the elements? There is no examples in the documentation that shows how to proceed after creating the processor

**Comment by wwkimball** [comment]
I'm sorry the abundant available documentation doesn't meet your standard.

I've provided an [extensive Wiki](https://github.com/wwkimball/yamlpath/wiki) with [a whole section entirely dedicated to interfacing with the library](https://github.com/wwkimball/yamlpath/wiki/Python-Library).  I have provided complete reference implementations of the capabilities of this library by way of [heavily documented command-line tools](https://github.com/wwkimball/yamlpath/wiki/Command-Line-(CLI)-Tools) and [their source files](https://github.com/wwkimball/yamlpath/tree/master/yamlpath/commands).  The API is thoroughly documented in classic form [within the library files themselves](https://github.com/wwkimball/yamlpath/tree/master/yamlpath).

What more do you need?

**Comment by fsmMLK** [comment]
I will check the documentation. I did not know some of the links you sent. Thanks

One question: say I have a processor X connected to yaml_data.

Is it possible to extract a node    myNode=X.get_nodes('path/to/node')

And create another processor, starting from myNode?  AFAIK myNode is a generator.

**Comment by wwkimball** [comment]
The yield from the generators is usually zero to many [NodeCoords](https://github.com/wwkimball/yamlpath/blob/master/yamlpath/wrappers/nodecoords.py) except when directly working against [Collectors](https://github.com/wwkimball/yamlpath/wiki/Segment:-Collectors), in which case you'll get a list of NodeCoords.

On their own, NodeCoords are not suitable as input to Processors.  It would however be technically possible -- even trivial -- to expose the `node` memvar as a read-only property and pass that to a new Processor.  That said, I'd need more knowledge about your use-case before suggesting that course of action.  You may find that a sole YAML Path may perform everything you need without the additional complexity of multiple Processors.

**Comment by wwkimball** [comment]
Since this has become an academic exploration rather than a bug report, I'm converting this thread to a discussion.

**Comment by marcelloromani** [comment]
The README lacks a simple example for the most dead-simple use case of yaml-path --search

* searching for e.g. limits.cpu in a K8s manifest
* searching for a literal in a YAML file

I appreciate that the YAML path syntax is obvious to advanced users - let alone the library developer - but the lack of an example for the most basic use case is definitely odd, and resonates with the OP's claim.
